### PR TITLE
feat: migrate transaction envelope to Mainsail (RLP + ECDSA)

### PR DESCRIPTION
**Pull request by shahin-hq**
## Summary
Closes: https://app.clickup.com/t/2570579/86e2fc2w4

- Migrate transaction envelope to RLP (Mainsail/EVM-compatible) - replace the legacy Ark v2 binary format with [nonce, gasPrice, gasLimit, to, value, data, v, r, s] RLP encoding
- Add hand-rolled RLP codec (rlp.go) - typed `RlpBytes/RlpBigInt/RlpList` `encode/decode`, no external dependency
- Add ABI codec (abi.go, contracts.go) - function-call encode/decode for the 11 known Mainsail contract signatures
- Replace Schnorr signing with recoverable ECDSA (secp256k1)
- Rework network configuration - chainId-based Network
- New `Transaction` struct - EVM-shaped fields (Nonce/GasPrice/GasLimit/To/Value/Data/V/R/S/...) replacing the old asset-based model
- Rewrite serializer/deserializer around the RLP envelope + EIP-155 v encoding
- Rewrite transaction signing/verification - `Sign`, `RecoverSender`, `Verify`, `GetHash`, `SigningHash`
- Harden ABI/RLP decoders against malformed input - bounded integer conversions, list-length validation, no more silent truncation or panics on bad wire data
- Trim dependencies - drop `btcsuite/btcd` and `ellemouton/schnorr` entirely; move to `decred/dcrd/dcrec/secp256k1/v4` directly; upgrade `golang.org/x/crypto`, `testify`
- Remove dead legacy-model code - enums.go, TransactionAsset/multisig types, and tests tied to the removed Schnorr/multisig builders
- Add end-to-end round-trip tests - sign → serialize → deserialize → verify, plus negative/edge cases

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: crypto/fixtures/identity.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
         "privateKey": "d8839c2432bfd0a67ef10a804ba991eabba19f154a3d707917681d45822a5712",
         "publicKey": "034151a3ec46b5670a682b0a63394f863587d1bc97483b1b6c70eb58e7f0aed192",
         "address": "0xb0FF9213f7226bBB72b84dE16af86e56f1f38B01",
-        "wif": "SGq4xLgZKCGxs7bjmwnBrWcT4C1ADFEermj846KC97FSv1WFD1dA"
+        "wif": "Ue7A6vSx7ewATPp2dA6UbJ8F39DbZwaHTqhD1MrhzmJqRJmvfZ6C"
```
Comment (ItsANameToo) [inline crypto/fixtures/identity.json:6]: is this intended?

File: crypto/message.go [hunk 2]
```diff
@@ -14,46 +14,43 @@ import (
 	"github.com/fatih/structs"
 )
 
+// SignMessage signs an arbitrary message with the private key derived from
+// passphrase.
+//
+// NOTE: this still hashes the message with plain sha256, carried over
+// unchanged from before the Mainsail migration. Ethereum's personal_sign
+// convention (keccak256("\x19Ethereum Signed Message:\n" + len(message) +
+// message)) is a separate, not-yet-made design change.
```
Comment (ItsANameToo) [inline crypto/message.go]: is the comment necessary?

File: crypto/abi.go [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+var (
+	ErrAbiInvalidAddress      = errors.New("abi: invalid address")
+	ErrAbiValueTooLarge       = errors.New("abi: value exceeds one word (32 bytes)")
+	ErrAbiNegativeUint        = errors.New("abi: uint256 must be non-negative")
+	ErrAbiUnexpectedEndOfData = errors.New("abi: unexpected end of data")
+	ErrAbiSelectorMismatch    = errors.New("abi: function selector does not match")
+	ErrAbiInvalidOffset       = errors.New("abi: dynamic value offset is invalid")
+)
+
+// AbiFunctionSelector returns the first 4 bytes of keccak256(signature), e.g.
```
Comment (ItsANameToo) [inline crypto/abi.go]: have a look at the comments, I don't think most of them add anything useful that you can't figure out directly from the code itself

**Review by ItsANameToo — COMMENTED** [review]
can you add an implementation to https://github.com/ArdentHQ/sdk-tester-scripts to see it in action @shahin-hq ? that will make testing easier too

File: crypto/fixtures/identity.json (hunk 1 above)
Comment (shahin-hq) [reply crypto/fixtures/identity.json:6]: Yes, https://github.com/ArkEcosystem/php-crypto/blob/b4b71f582c8d6202f1f51d9a889489a74581fccf/src/Networks/Testnet.php#L15

- Legacy Devnet/Mainnet both used Wif: 170 (0xaa).
- Mainsail's Mainnet/Testnet both use Wif: 186 (0xba).

File: crypto/abi.go (hunk 3 above)
Comment (shahin-hq) [reply crypto/abi.go]: @ItsANameToo I kept only the useful comments, please, have a look.

**Review by ItsANameToo — APPROVED** [review]
